src/autoload/legacy-autoloader.php

```diff
@@ -71,12 +72,22 @@ function glpi_autoload($classname)
 
     // Legacy class path, e.g. `PluginMyPluginFoo` -> `plugins/myplugin/inc/foo.class.php`
     $legacy_path          = sprintf('%s/inc/%s.class.php', $plugin_path, str_replace('\\', '/', strtolower($plugin_class)));
-    // PSR-4 styled path for class without namespace, e.g. `PluginMyPluginFoo` -> `plugins/myplugin/src/PluginMyPluginFoo.php`
+
+    // PSR-4 styled path for full plugin classes (with or without namespaces),
+    // Resolves `PluginMyPluginFoo` -> `plugins/myplugin/src/PluginMyPluginFoo.php`
+    // Will also resolve "GlpiPlugin\MyPlugin\MyClass" -> "plugins/myplugin/src/GlpiPlugin/MyPlugin/MyClass.php"
     $psr4_styled_path     = sprintf('%s/src/%s.php', $plugin_path, str_replace('\\', '/', $classname));
+
+    // PSR-4 styled path for class with proper namespace configuration,
+    // will resolve "GlpiPlugin/MyPlugin/MyClass.php" to "plugins/myplugin/src/MyClass.php"
+    $psr4_shortened_path  = sprintf('%s/src/%s.php', $plugin_path, str_replace('\\', '/', $plugin_class));
```

Comment:
This is weird, I don't understand why this specific case isn't already covered by the existing code.
Could you create a test that fails without your changes to demonstrate why this new code is needed ?

I've tried with this:
![image](https://github.com/user-attachments/assets/2f7fd23f-f1f9-4301-8c59-bfc83ff7de02)

With the matching file:
![image](https://github.com/user-attachments/assets/91c75de0-a40b-46db-b3ae-3c0d801b5c38)

But the test works fine on my end (main branch, so without your changes).